packages/consensus/source/proposer-picker.ts

```diff
@@ -13,10 +14,9 @@ export class ProposerPicker implements Contracts.Consensus.IProposerPicker {
 	private validatorIndexMatrix: Array<number> = [];
 
 	public async onCommit(committedBlock: Contracts.Crypto.ICommittedBlock): Promise<void> {
-		const { activeValidators } = this.configuration.getMilestone();
-
-		const height = committedBlock.block.data.height;
-		if (this.validatorIndexMatrix.length === 0 || height % activeValidators === 0) {
+		const { height } = committedBlock.block.data;
+		if (this.validatorIndexMatrix.length === 0 || Utils.roundCalculator.isNewRound(height, this.configuration)) {
```

Comment:
New matrix should be recalculated when last block in the round is committed.

```suggestion
		if (this.validatorIndexMatrix.length === 0 || Utils.roundCalculator.isNewRound(height + 1, this.configuration)) {
```